src/java/org/apache/cassandra/cql3/statements/QualifiedStatement.java

```diff
@@ -78,4 +78,50 @@ public String toString()
     {
         return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
     }
+
+    public static abstract class CompositeQualifiedStatement extends QualifiedStatement
```

Comment:
Shouldn't it implement `CompositeCQLStatement` ?

Reply:
nit: Given most usage looks like `QualifiedStatement.CompositeQualifiedStatement`, I'd consider just naming this `Composite`, so it's more like `QualifiedStatement.Composite`...but not super important. Fine w/ whatever you go with.

Reply:
`QualifiedStatement extends CQLStatement.Raw`
`CompositeCQLStatement extends CQLStatement`

so, can't...

Reply:
switched to `Composite`, debated this... if we pull out from this class should rename I guess